src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,169 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MagicWandService {
+  /**4-Way floodfill (left, right, up, down):
+   * Return set of coordinates(formatted as a 1-D array).
+   * Coordinates correspond to pixels considered part of the mask.
+   */
+  /* tslint:disable */
+  floodfill(imgData: ImageData, xCoord: number, yCoord: number,
+            tolerance: number): Set<number> {
+    // Use basic queue to immitate recursive approach of the stack
+    // possible TODO: set array size after mvp to optimize heap memory access
```

Comment:
This comment still leaves some questions in my mind about what these variables mean. I'd suggest spelling it out:

We will store a queue of pixels that we need to visit in \`visit\`.
We will store the already-visited pixels in \`visited\`.